src/shims/posix/fs.rs

```diff
@@ -827,9 +847,25 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
         this.assert_target_os("macos", "stat");
-        this.check_no_isolation("`stat`")?;
+
+        let path_scalar = this.read_pointer(path_op)?;
+        let path = this.read_path_from_c_str(path_scalar)?.into_owned();
+
+        // Reject if isolation is enabled.
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("`stat`", reject_with)?;
+            let eacc = this.eval_libc("EACCES")?;
+            this.set_last_error(eacc)?;
+            return Ok(-1);
+        }
+
         // `stat` always follows symlinks.
-        this.macos_stat_or_lstat(true, path_op, buf_op)
+        let metadata = match FileMetadata::from_path(this, &path, true)? {
+            Some(metadata) => metadata,
+            None => return Ok(-1),
+        };
+
+        this.macos_stat_write_buf(metadata, buf_op)
```

Comment:
`macos_stat_write_buf` is unused now and can be removed, right?

Reply:
I guess you meant `macos_stat_or_lstat`. Right, I forgot to delete it and I  guess rustc didn't complain because it's a trait function? Thanks for pointing out.

Reply:
Ah yes I dd, sorry for the confusion.